### PR TITLE
[iOS][BarCodeScanner] Add support for Codabar

### PR DESCRIPTION
**Pull request by 7nohe**
# Why

Added support for Codabar barcode type, since iOS SDK does not support it.

# How

I've added zxing's`ZXCodaBarReader` for scanning Codebar barcode type.

# Test Plan

- Ran the `bare-expo` project on a physical device, scanned a Codabar barcode (e.g. image from https://en.wikipedia.org/wiki/Codabar)

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by bbarthec — CHANGES_REQUESTED** [review]
Key @7nohe 👋 Thank you for this PR 🤩 
The code looks good, but CI is failing (https://github.com/expo/expo/runs/5625155998?check_suite_focus=true), because `AVMetadataObjectTypeCodabarCode` is not available on the iOS 🤔 I've checked the documentation and it seems this variable is added in iOS 15.4 (https://developer.apple.com/documentation/avfoundation/avmetadataobjecttypecodabarcode/), so we'll need to add some compatibility layer for older iOS versions (check [@availbale(..., ...) syntax](https://developer.apple.com/documentation/swift/objective-c_and_c_code_customization/marking_api_availability_in_objective-c)).

**Comment by 7nohe** [comment]
Thank you for the review @bbarthec!
I fixed the code to support older iOS versions by using `@available()` syntax.
It seems to be working fine on the `bare-expo`, but not sure about the tests 😭

File: packages/expo-barcode-scanner/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 ### 🎉 New features
 
+- Added support for Codabar barcode type on iOS. ([#16703](https://github.com/expo/expo/pull/16703) by [@7nohe](https://github.com/7nohe))
```
Comment (bbarthec) [inline packages/expo-barcode-scanner/CHANGELOG.md:9]:
```suggestion
-  On iOS 15.4+ added support for `Codabar` barcode type. ([#16703](https://github.com/expo/expo/pull/16703) by [@7nohe](https://github.com/7nohe))
```

**Review by bbarthec — APPROVED** [review]
It looks fine now 👍 
Thank you for taking care of this 😁

**Pull request merged**